pallets/creditcoin/src/types/collect_coins.rs

```diff
@@ -131,7 +131,7 @@ where
 
 #[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
 #[allow(clippy::upper_case_acronyms)]
-pub enum BurnDetails {
+pub enum TokenContract {
```

Comment:
I kinda prefer `BurnDetails` actually. `TokenContract` makes it sound like the `ExternalAddress` here is the address of the contract (at least to me).